### DISCUSSION TITLE
Troubleshooting and Warning W31 on Deye

### DISCUSSION THREAD

**Discussion by cfi700** [ArminJo/JK-BMSToPylontechCAN#3] [General]
First of all, thanks for the great project.

While rebuilding (with Arduino Nano Clone) I noticed the following problems:
PIN SCL is not PIN A3 but PIN A5.
PIN Rx is not PIN 1 as specified but PIN 2.

If you use the cable from the RS485 adapter of the JK-BMS and follow the labeling on the board, you have to swap the lines for RX and TX on the nano.

The BigNumbers are not always displayed correctly, mostly when the current changes from two digits (e.g. 31.5 A) to one digit (e.g. 7.9 A).

The Inverter (SUN-10K-SG04LP03-EU) displays the warning "W31 Battery_comm_warn" after a short time. Nevertheless, correct data is displayed on the Deye.

**Comment by ArminJo** [comment]
Thanks for doing the QS 👍 
These and many other display bugs will be fixed today.

**Reply by cfi700** [reply]
Hello, I have found a solution for the warning W31.
When transmitting the battery voltage, the value in the CAN frame is shifted by one decimal place.
After the change:
FrameData.Voltage100Millivolt = JKComputedData.BatteryVoltage10Millivolt;// / 10;
the voltage value is transferred correctly and the warning disappears.

**Reply by ArminJo** [reply]
Thanks 🥇 
I converted this discussion to the issue #5 and will delete this discussion soon.
